src/plugins/intel_cpu/src/nodes/kernels/x64/random_uniform.cpp

```diff
@@ -626,10 +640,675 @@ void RandomUniform<isa>::tail(const std::vector<Vmm>& vmm_dst) {
+        vmovups(v_divisor, ptr[r64_aux]);
+
+        // Note: two times too many values in Zmm
+        BROADCAST_MERSENNE_PARAM(vpbroadcastw, v_range,     r64_aux, range_ptr)
+        BROADCAST_MERSENNE_PARAM(vpbroadcastw, v_min,       r64_aux, min_ptr)
+    } else if (m_jcp.out_data_type == element::i32) {
+        BROADCAST_MERSENNE_PARAM(vpbroadcastd, v_range, r64_aux, range_ptr)
+        BROADCAST_MERSENNE_PARAM(vpbroadcastd, v_min,   r64_aux, min_ptr)
+    } else if (m_jcp.out_data_type == element::i64) {
+        // Same as in Philox - in scope of i64 enabling
+        OPENVINO_THROW("RandomUniform kernel does not support precision ", m_jcp.out_data_type, " for ", x64::get_isa_info());
+    } else {
+        OPENVINO_THROW("RandomUniform kernel does not support precision ", m_jcp.out_data_type, " for ", x64::get_isa_info());
+    }
+}
+
+template <x64::cpu_isa_t isa> // Works for AVX2, SSE41
+void MersenneTwisterGenerator<isa>::initVectors() {
+    const auto r64_aux = getReg64();
+
+    v_min = getVmm();
+    v_range = getVmm();
+    v_state = getVmm();
+    v_result = getVmm();
+    v_const_1 = getVmm();
+    v_const_2 = getVmm();
+
+    // Initialize state
+    uni_vmovdqu(v_state, ptr[r64_state]);
+
+    // Initialize constants.
+    INIT_8_ELEM_T_ARR(const_1, MT_CONST_1, r64_aux, uint32_t);
+    uni_vmovdqu(v_const_1, ptr[r64_aux]);
+
+    INIT_8_ELEM_T_ARR(const_2, MT_CONST_2, r64_aux, uint32_t);
+    uni_vmovdqu(v_const_2, ptr[r64_aux]);
+
+    if (m_jcp.out_data_type == element::f32) {
+        v_mask = getVmm();
+        v_divisor = getVmm();
+
+        INIT_8_ELEM_T_ARR(mask, static_cast<uint32_t>((1 << 24) - 1), r64_aux, uint32_t);
+        uni_vmovups(v_mask, ptr[r64_aux]);
+
+        INIT_8_ELEM_T_ARR(divisor, static_cast<float>(1.0f / (1 << 24)), r64_aux, float);
+        uni_vmovups(v_divisor, ptr[r64_aux]);
+
+        mov(r64_aux, ptr[r64_params + GET_MERSENNE_OFFSET(range_ptr)]);
+        uni_vpbroadcastd(v_range, ptr[r64_aux]);
+
+        mov(r64_aux, ptr[r64_params + GET_MERSENNE_OFFSET(min_ptr)]);
+        uni_vpbroadcastd(v_min, ptr[r64_aux]);
+    } else if (m_jcp.out_data_type == element::i32) {
+        mov(r64_aux, ptr[r64_params + GET_MERSENNE_OFFSET(range_ptr)]);
+        uni_vpbroadcastd(v_range, ptr[r64_aux]);
+
+        mov(r64_aux, ptr[r64_params + GET_MERSENNE_OFFSET(min_ptr)]);
+        uni_vpbroadcastd(v_min, ptr[r64_aux]);
+    } else {
+        OPENVINO_THROW("RandomUniform kernel does not support precision ", m_jcp.out_data_type, " for ", x64::get_isa_info());
+    }
+}
+
+template <x64::cpu_isa_t isa>
+void MersenneTwisterGenerator<isa>::process() {
+    // Generate random numbers
+    generateRandomNumbers();
+
+    // Convert to output type, store result
+    convertToOutputTypeMersenne();
+
+    // Store results
+    storeResults();
+}
+
+template <>
+void MersenneTwisterGenerator<x64::avx512_core>::generateRandomNumbers() {
+    const auto v_aux = getVmm();
+
```

Comment:
```
    vpsrld(v_aux, v_state, 11);
    vpxord(v_result, v_state, v_aux);

    vpslld(v_aux, v_result, 7);
    vpandd(v_aux, v_aux, v_const_1);
    vpxord(v_result, v_result, v_aux);

    vpslld(v_aux, v_result, 15);
    vpandd(v_aux, v_aux, v_const_2);
    vpxord(v_result, v_result, v_aux);

    vpsrld(v_aux, v_result, 18);
    vpxord(v_result, v_result, v_aux);
```

Reply:
The same for SSE41, AVX2

Reply:
Fixed!

Reply:
Update: It fails for uni_vmovdqu for AVX512 Zmm->Zmm copy. I tried adding it to jit_generator but I got "EVEX is invalid" error that I could not resolve